src/services/FDC3/channelClient.ts

```diff
@@ -53,12 +53,19 @@ export default class C implements Channel {
 		}
 
 		if (this.id == "global") {
-			const routerHandler: StandardCallback = (err, response) => {
-				//prevent message loops
-				if (response.data.source != this.#FSBL.Clients.WindowClient.getWindowIdentifier().windowName) {
-					//delete response.data.source // delete non standard source field we added
-					theHandler(response.data.context);
+			const routerHandler: StandardCallback = (err: any, response: { data: { source: string; context: Context; }; }) => {
+
+				const { source, context } = response.data
```

Comment:
Is there some reason to transmit the source instead of using the source from Finsemble's message headers?

Reply:
After discussing with Sidd and a quick test, you can do:

Router:
```
FSBL.Clients.RouterClient.addListener("testKris2", (err,response) => {
    if (response.originatedHere()) {
        console.log("It did");
    } else {
        console.log("It didn't");
    }
    console.log(response);
});
```
Linker:
```
FSBL.Clients.LinkerClient.subscribe("testKris2", (response,envelope) => {
    if (envelope.originatedHere()) {
        console.log("It did");
    } else {
        console.log("It didn't");
    }
    console.log(envelope);
});
```

Reply:
From Slack thread:
Watson:
> So I ran the code last night and indeed it worked. I then did the same in the FDC3 code and the results were not what I expected. originatedHere() was returning false even if it originated in the window. Is this due to the way that broadcast works?
> This is the header value(if it helps)

```
{
  "origin": "RouterClient.FDC3",
  "type": "transmit",
  "channel": "group1",
  "originIncomingTransportInfo": {
    "transportID": "SharedWorker",
    "port": 17
  },
  "lastClient": "RouterClient.Welcome Component-86-1086-Finsemble",
  "incomingTransportInfo": {
    "transportID": "SharedWorker",
    "port": 1
  }
}
```
-------

Agreed that using originatedHere() won't work for FDC3 hence using the source.